python/cuml/internals/base.pyx

```diff
@@ -57,6 +56,9 @@ from cuml.internals.mixins import TagsMixin
 cp_ndarray = gpu_only_import_from('cupy', 'ndarray')
```

Comment:
For these GPU-only imports, would it make sense to move them into the `IF GPUBUILD` clause? Fine to leave it for a follow-up if so.

Reply:
The `IF GPUBUILD` bits just need to surround C++/cdef structures that get cythonized to become symbol references in the compiled objects, so there is no need to surround 99% of the Python bits, just C++/cython CUDA specific things. I think it's better to try to minimize the things inside those `IF` parts in case we need to restructure things in the future, what do you think?

Reply:
Yes, agree 100%.